reconcile/utils/terrascript_aws_client.py

```diff
@@ -5152,3 +5158,68 @@ def populate_tf_resource_rosa_authenticator(self, spec):
         tf_resources.append(waf_acl_association_resource)
 
         self.add_resources(account, tf_resources)
+
+    def populate_tf_resource_rosa_authenticator_vpce(self, spec):
+        account = spec.provisioner_name
+        identifier = spec.identifier
+        common_values = self.init_values(spec)
+        tf_resources = []
+        self.init_common_outputs(tf_resources, spec)
+
+        vpc_id = common_values.get("vpc_id")
+        subnet_ids = common_values.get("subnet_ids")
+        vpce_security_group_rule_common_args = common_values.get(
+            "vpce_security_group_rule_common_properties", None
+        )
+        vpc_endpoint_args = common_values.get("vpc_endpoint_properties", None)
+
+        # VPC ENDPOINT NETWORK MODULE SECTION
+        # SG
+        aws_security_group_resource = aws_security_group(
+            "api_gw_vpce",
+            name=f"ocm-{identifier}-api-gateway-vpce-sg",
+            description="Control access to the API Gateway VPC endpoint",
+            vpc_id=vpc_id,
+            tags={"Name": f"ocm-{identifier}-api-gateway-vpce-sg"},
+        )
+        tf_resources.append(aws_security_group_resource)
+
+        # SG RULES
+        aws_security_group_rule_inbound_resource = aws_security_group_rule(
+            "vpce_inbound",
+            type="ingress",
+            security_group_id=f"${{{aws_security_group_resource.id}}}",
+            cidr_blocks=["10.0.0.0/8"],
```

Comment:
This will need to match the VPC CIDR blocks

Reply:
This is how the prototype is written, we're just implementing it. Example with a data source / variable input?

Reply:
https://coreos.slack.com/archives/C027UG79RR9/p1658263031244359?thread_ts=1658253407.763229&cid=C027UG79RR9

I'm interpreting this to mean that the current value in the code is fine.

Reply:
Yes, `10.0.0.0/8` should be ok for the current stage/production VPCs, but future VPCs could have different CIDRs.